htlcswitch/switch.go

```diff
@@ -513,18 +513,10 @@ func (s *Switch) handleChanelClose(req *ChanClose) {
 	return
 }
 
-// htlcForwarder is responsible for optimally forwarding (and possibly
-// fragmenting) incoming/outgoing HTLCs amongst all active interfaces and their
-// links. The duties of the forwarder are similar to that of a network switch,
-// in that it facilitates multi-hop payments by acting as a central messaging
-// bus. The switch communicates will active links to create, manage, and tear
-// down active onion routed payments. Each active channel is modeled as
-// networked device with metadata such as the available payment bandwidth, and
-// total link capacity.
+// handleControl...
 //
 // NOTE: This MUST be run as a goroutine.
-func (s *Switch) htlcForwarder() {
-	defer s.wg.Done()
+func (s *Switch) handleControl() {
```

Comment:
What deadlock? Can you add some detail to the commit's message explaining the scenario that caused you to make this change? Thanks!